component/metrics/mutate/mutate.go

```diff
@@ -101,19 +102,21 @@ func (c *Component) Update(args component.Arguments) error {
 // series. This is a blocker for releasing a production-grade  of the metrics.mutate
 // component.
 func (c *Component) Receive(ts int64, metricArr []*metrics.FlowMetric) {
-	app := c.appendable.Appender(context.Background())
+	c.mut.Lock()
+	defer c.mut.Unlock()
```

Comment:
This should probably be a RLock/RUnlock since many things might be sending metrics at the same time